### PR TITLE
qa: make the xfstests tmp directories to be readable and excutable

### PR DESCRIPTION
**Pull request by lxbsz**
The user namespace test case will excute some commands as the other user. And the temporary direcotry have now the rx permission for other users by default and then will fail the tests.

This will chmod by making it to be readable and excutable.

Fixes: https://tracker.ceph.com/issues/58760

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by lxbsz** [comment]
Both the relevant [qa acls tests](https://pulpito.ceph.com/xiubli-2023-02-23_03:24:53-fs:functional-wip-fscrypt-20230215-0834-distro-default-smithi) and [qa fscrypt tests](https://pulpito.ceph.com/xiubli-2023-02-23_03:24:22-fs:fscrypt-wip-fscrypt-20230215-0834-distro-default-smithi) are running.

**Comment by lxbsz** [comment]
> Both the relevant [qa acls tests](https://pulpito.ceph.com/xiubli-2023-02-23_03:24:53-fs:functional-wip-fscrypt-20230215-0834-distro-default-smithi) and [qa fscrypt tests](https://pulpito.ceph.com/xiubli-2023-02-23_02:38:54-fs:fscrypt-wip-fscrypt-20230215-0834-distro-default-smithi) are running.

Both passed.

File: qa/tasks/cephfs/xfstests_dev.py [hunk 1]
```diff
@@ -187,7 +195,7 @@ def write_ceph_exclude(self):
         # These tests will fail or take too much time and will
         # make the test timedout, just skip them for now.
         xfstests_exclude_contents = dedent('''\
-            {c}/001 {g}/003 {g}/020 {g}/075 {g}/317 {g}/538 {g}/531
```
Comment (dparmar18) [inline qa/tasks/cephfs/xfstests_dev.py:190]: is this removal intentional?

Comment (lxbsz) [reply qa/tasks/cephfs/xfstests_dev.py:190]: Yeah. Since the bug has been fixed this will add the test back. I will update the commit comment about this.

**Comment by lxbsz** [comment]
Only updated the commit comment since last tests.

File: qa/tasks/cephfs/xfstests_dev.py (hunk 1 above)
Comment (dparmar18) [reply qa/tasks/cephfs/xfstests_dev.py:190]: thanks

File: qa/tasks/cephfs/xfstests_dev.py [hunk 2]
```diff
@@ -61,13 +61,21 @@ def get_repos(self):
...
         self.mount_a.run_shell(['git', 'clone', remoteurl, '--depth', '1',
                                 self.xfstests_repo_path])
 
         if self.install_xfsprogs:
             remoteurl = 'https://git.ceph.com/xfsprogs-dev.git'
             self.xfsprogs_repo_path = self.mount_a.client_remote.mkdtemp(suffix=
                                                                 'xfsprogs-dev')
+            # Make the xfsprogs_repo_path to be readable and excutable for
+            # other users, this will allow the xfstests to run the user namespace
+            # test cases.
```
Comment (dparmar18) [inline qa/tasks/cephfs/xfstests_dev.py:77]: both the comments are the same, maybe have just one comment to reduce verbosity? Else LGTM

Comment (lxbsz) [reply qa/tasks/cephfs/xfstests_dev.py]: Let's keep it to make the code more easy to read.
I have updated the comments in the code as you mentioned in G-chat. @dparmar18 

Thanks.

**Comment by lxbsz** [comment]
Only updated the comments in code since last tests.

File: qa/tasks/cephfs/xfstests_dev.py (hunk 2 above)
Comment (dparmar18) [reply qa/tasks/cephfs/xfstests_dev.py]: thanks

**Review by dparmar18 — APPROVED** [review]

File: qa/tasks/cephfs/xfstests_dev.py (hunk 1 above)
Comment (rishabh-d-dave) [inline qa/tasks/cephfs/xfstests_dev.py:190]: Technically, it would better to have a separate commit for this change since it doesn't fit with rest of the changes in this commit.

**Review by rishabh-d-dave — APPROVED** [review]
I also suggest some improvements to the commit message -
```
The user namespace test case will execute some commands as the other
user. The temporary directory now has the rx permission for
other users by default. The tests then will fail otherwise.

Add back the generic/317 test.
```

Copying old commit message here for reference -

```
The user namespace test case will excute some commands as the other
user. And the temporary direcotry have now the rx permission for
other users by default and then will fail the tests.

This will chmod by making it to be readable and excutable.
```

File: qa/tasks/cephfs/xfstests_dev.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/cephfs/xfstests_dev.py:255]: Done and fixed them all.

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by lxbsz** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Comment by lxbsz** [comment]
> jenkins test make check

The `make check` failure was newly introduced and has been fixed in https://github.com/ceph/ceph/pull/50268.

**Comment by dparmar18** [comment]
> > jenkins test make check
> 
> The `make check` failure was newly introduced and has been fixed in #50268.

without that patch, this PR would still fail at make check, maybe merge both PRs?

**Comment by lxbsz** [comment]
> > > jenkins test make check
> > 
> > 
> > The `make check` failure was newly introduced and has been fixed in #50268.
> 
> without that patch, this PR would still fail at make check, maybe merge both PRs?

Yeah, sounds good to me.

**Comment by lxbsz** [comment]
> > > jenkins test make check
> > 
> > 
> > The `make check` failure was newly introduced and has been fixed in #50268.
> 
> Which commit introduced the failure?

This PR https://github.com/ceph/ceph/pull/45669, before why that PR didn't fail it should be the `ext_num_fwd` and `ext_num_retry` in `head` were initialized as `0` randomly.

**Comment by lxbsz** [comment]
Rebased it.
